### ISSUE TITLE
NativeViewHierarchyManager: Unable to update properties for view tag 12187

### ISSUE THREAD

**Issue by OrenMe** [software-mansion/react-native-reanimated#3860]
### Description

App crashes and I see this appearing in our Sentry hitting many end users.
```
E/unknown:NativeViewHierarchyManager: Unable to update properties for view tag 12187
    com.facebook.react.uimanager.IllegalViewOperationException: ViewManager for tag 12187 could not be found.
    
        at com.facebook.react.uimanager.NativeViewHierarchyManager.resolveViewManager(NativeViewHierarchyManager.java:111)
        at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:138)
        at com.facebook.react.uimanager.UIImplementation.synchronouslyUpdateViewOnUIThread(UIImplementation.java:305)
        at com.swmansion.reanimated.NodesManager.updateProps(NodesManager.java:588)
        at com.swmansion.reanimated.NativeProxy.updateProps(NativeProxy.java:167)
        at com.swmansion.reanimated.NativeProxy$AnimationFrameCallback.onAnimationFrame(Native Method)
        at com.swmansion.reanimated.NodesManager.onAnimationFrame(NodesManager.java:273)
        at com.swmansion.reanimated.NodesManager.access$000(NodesManager.java:65)
        at com.swmansion.reanimated.NodesManager$1.doFrameGuarded(NodesManager.java:170)
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:964)
        at android.view.Choreographer.doCallbacks(Choreographer.java:790)
        at android.view.Choreographer.doFrame(Choreographer.java:721)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:951)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

### Steps to reproduce

Issue is happening mainly on luncher apps when clicking on MENU button, hard to give proper steps to reproduce as this is mainly coming from Sentry alerts
This could be related to fact that app goes to background when this happens

### Snack or a link to a repository

Private repo

### Reanimated version

2.6.0

### React Native version

react-native-tvos@0.66.3-1

### Platforms

Android

### JavaScript runtime

None

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Release mode

### Device

Real device

### Acknowledgements

Yes

**Comment by sweatherall** [comment]
Also getting this issue on Android after upgrading dependencies:
`react native: 0.71.3`
`react-native-reanimated: 2.14.4`
Causing the app to crash.
Full error log from android studio:
```
Unable to update properties for view tag 7679
com.facebook.react.uimanager.IllegalViewOperationException: ViewManager for tag 7679 could not be found.
at com.facebook.react.uimanager.NativeViewHierarchyManager.resolveViewManager(NativeViewHierarchyManager.java:111)
at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:138)
at com.facebook.react.uimanager.UIImplementation.synchronouslyUpdateViewOnUIThread(UIImplementation.java:281)
at com.swmansion.reanimated.NodesManager.updateProps(NodesManager.java:590)
at com.swmansion.reanimated.NativeProxy.updateProps(NativeProxy.java:185)
at com.swmansion.reanimated.NativeProxy$AnimationFrameCallback.onAnimationFrame(Native Method)
at com.swmansion.reanimated.NodesManager.onAnimationFrame(NodesManager.java:275)
at com.swmansion.reanimated.NodesManager.access$000(NodesManager.java:67)
at com.swmansion.reanimated.NodesManager$1.doFrameGuarded(NodesManager.java:172)
at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1035)
at android.view.Choreographer.doCallbacks(Choreographer.java:845)
at android.view.Choreographer.doFrame(Choreographer.java:775)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7842)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by Marvedog** [comment]
Hello! I am also experiencing this issue in our app.

* `react-native: 0.68.5`
* `react-native-reanimated: 2.9.1`

I am able to consistently reproduce it, but it is quite difficult to share the repro here due to the specific scenario it is triggered in. I can share some details, and maybe someone can help me with some ideas to produce a repro I can share here. 

We have a health app which continually streams sensor data from external BLE sensors for 10 minutes. After the 10 minutes of streaming we do the following quite intensive operations:
* We compute some computationally expensive statistics on the JS layer
* We disconnect the BLE sensors
* We then trigger an some `reanimated` animation from the streaming screen to a statistics screen

During the transition from the streaming screen to the statistics screen the exact error reported by @sweatherall occurs. The weird thing is that I am not able to reproduce it without computing the full sequence of steps outlined above. We have implemented the `reanimated` animations in a separate component library with storybook, but the animation works totally fine here. If anyone has any ideas that would be helpful! I will try to create a repro in our component library and I can possibly provide temporary access to someone from the `reanimated` team if that is useful :)

**Comment by Latropos** [comment]
Closing since we are unable to solve this issue due to lack of reproduction. However I believe that the problem described is already fixed (I remember similar bugs with FlatList hat got already resolved)

**Comment by piaskowyk** [comment]
Fixed with https://github.com/software-mansion/react-native-reanimated/pull/5767, changes will be available in the next release - 3.12.0
